metadata/src/main/java/org/apache/kafka/image/LocalReplicaChanges.java

```diff
@@ -26,19 +26,22 @@
 
 public final class LocalReplicaChanges {
     private final Set<TopicPartition> deletes;
-    private final Map<TopicPartition, PartitionInfo> leaders;
+    private final Map<TopicPartition, PartitionInfo> electedLeaders;
+    private final Map<TopicPartition, PartitionInfo> updatedLeaders;
```

Comment:
Does it make sense to change these names to tpToPartitionEpochs and tpToLeaderEpochs? I can anticipate this naming being confusing for a person reading the code for the first time given that what classifies as each is defined in https://github.com/apache/kafka/pull/14489/files#diff-be8b1b8ad296c48bbdc3df55fdb859881f150ceadd0959ebf02fb3caac13ee5aR146-R151

Reply:
I struggled with naming this quite a bit 😅 . I was also wondering if I should make it clearer abou leader epoch changes vs partition epoch changes. The thing that is tricky is that the map only contains the leaders that experienced the changes (not followers) so I also wanted to make that clear. I will also think on that some more.

Reply:
I pushed changes for the simpler fixes and will continue to think on this one.